### PR TITLE
Add More examples

### PR DESCRIPTION
**Pull request by ResetXD**
#248 
interactions for cogs with better examples

### CODE REVIEW

File: examples/interactionsBot/cogs/slashGroup_cog.py
Comment (Lulalaby) [inline examples/interactionsBot/cogs/slashGroup_cog.py:33]: More documentation for user

File: examples/interactionsBot/cogs/button_cog.py
Comment (Lulalaby) [inline examples/interactionsBot/cogs/button_cog.py:82]: More documentation for user

File: examples/interactionsBot/cogs/button_cog.py
Comment (Lulalaby) [inline examples/interactionsBot/cogs/button_cog.py:19]: More documentation for user

**Review by Lulalaby — CHANGES_REQUESTED** [review]
And class names maybe camel case

**Comment by Lulalaby** [comment]
But anyway, thanks!

**Comment by ResetXD** [comment]
@Lulalaby updated with more documentation please review and tell if it needs any changes

**Review by Lulalaby — DISMISSED** [review]

**Comment by ResetXD** [comment]
> The cog loading should also be static, not os.listdir() since many people consider that to not be a good standard practice.

this have became the more "go-to" way of importing cogs nowerdays.

and the examples highlights all of the interactions possible in pycord.

> It would be much more effective to simply show a simple example of a cog, maybe with one of each type of application command. 
i personally feel that simple examples wont help the user understand the complex part of the interaction.

@BobDotCom

**Comment by nerdguyahmad** [comment]
As Bob mentioned above, I agree with that, This example does way too much stuff and is too vague. Examples are meant to just provide a very basic overview to users to give them a quickstart.

**Comment by ResetXD** [comment]
@nerdguyahmad @BobDotCom check and tell if it needs further revision

**Comment by BobDotCom** [comment]
I still feel like these examples are far too advanced. For an example showcasing usage of cogs, I feel it would make sense to simply have one cog, with one or two slash commands, and one of each type of context menu. It probably shouldn't even have components, since those aren't particularly affected by using cogs.

**Comment by Lulalaby** [comment]
> I still feel like these examples are far too advanced. For an example showcasing usage of cogs, I feel it would make sense to simply have one cog, with one or two slash commands, and one of each type of context menu. It probably shouldn't even have components, since those aren't particularly affected by using cogs.

Agreeing with that.
Advanced examples should be covered in external sources.

**Comment by nerdguyahmad** [comment]
^ that

**Comment by ResetXD** [comment]
alr i will try to further simplify the examples.

**Comment by ResetXD** [comment]
@Lulalaby @BobDotCom @nerdguyahmad

**Comment by Lulalaby** [comment]
@ResetXD please fix the code spell error https://github.com/Pycord-Development/pycord/runs/4294735196?check_suite_focus=true

**Comment by ResetXD** [comment]
@Lulalaby

**Comment by ResetXD** [comment]
@Lulalaby i found a way better to manage SlashCommandGroup in cogs or somewhat cogs i will implement that before

**Comment by ResetXD** [comment]
check and tell, i tested the code and it works fine.

**Comment by BobDotCom** [comment]
> @Lulalaby i found a way better to manage SlashCommandGroup in cogs or somewhat cogs i will implement that before

#462 will rework groups. Let's wait until that is merged and use the new syntax.

**Review by RPSMain — CHANGES_REQUESTED** [review]
should've gotten then all

File: examples/interactionsBot/cogs/slashOption_cog.py
Comment (ResetXD) [reply examples/interactionsBot/cogs/slashOption_cog.py:28]: unnecessary

File: examples/interactionsBot/cogs/slashGroup_cog.py
Comment (ResetXD) [reply examples/interactionsBot/cogs/slashGroup_cog.py:29]: unnecessary

File: examples/interactionsBot/cogs/slashOption_cog.py
Comment (ResetXD) [reply examples/interactionsBot/cogs/slashOption_cog.py:13]: unnecessary

**Comment by ResetXD** [comment]
@RPSMain replacing client with bot is not useful so why a change request.
many people uses client, few people also uses bot as the variable so why the change.

File: examples/interactionsBot/cogs/slashGroup_cog.py
Comment (RPSMain) [reply examples/interactionsBot/cogs/slashGroup_cog.py:29]: it is? it makes the examples follow pep8

**Comment by RPSMain** [comment]
> @RPSMain replacing client with bot is not useful so why a change request. many people uses client, few people also uses bot as the variable so why the change.

Consistency you are subclassing Bot so it should be Bot not Client.

**Comment by ResetXD** [comment]
also a space before ctx or any other function argument is irrelevant to pep 8

**Comment by RPSMain** [comment]
> also a space before ctx or any other function argument is irrelevant to pep 8

https://www.python.org/dev/peps/pep-0008/#variable-annotations

**Comment by ResetXD** [comment]
> Consistency you are subclassing Bot so it should be Bot not Client.

what about the file name, **bot.py** shouldn't it be called **main.py** then.

**Comment by RPSMain** [comment]
> what about the file name, **bot.py** shouldn't it be called **main.py** then.

if you want to.

**Comment by ResetXD** [comment]
> if you want to.

alr will fix all again ig.

**Comment by RPSMain** [comment]
@ResetXD have you linted the examples?

**Comment by Lulalaby** [comment]
> replacing client with bot is not useful so why a change request.

We don't support self-botting, which would be client. 
So for this purpose bot is better.

**Comment by BobDotCom** [comment]
> > replacing client with bot is not useful so why a change request.
> 
> We don't support self-botting, which would be client. 
> So for this purpose bot is better.

Client is used for bots, just not for slash commands.

**Review by Lulalaby — APPROVED** [review]

**Comment by RPSMain** [comment]
@ResetXD this should fix the errrors https://github.com/ResetXD/pycord/pull/1

**Review by RPSMain — APPROVED** [review]

**Pull request merged**
